File: app/src/main/java/com/bignerdranch/stockwatcher/model/service/StockInfoResponse.java
package com.bignerdranch.stockwatcher.model.service;

import com.google.gson.annotations.SerializedName;

import lombok.Data;

@Data
public class StockInfoResponse {

    @SerializedName("Status")
    private String status;

    @SerializedName("Name")
    private String name;

    @SerializedName("LastPrice")
    private double lastPrice;

    @SerializedName("Change")
    private double change;

    @SerializedName("ChangePercent")
    private double changePercent;

    @SerializedName("MarketCap")
    private long marketCap;

    @SerializedName("Volume")
    private long volume;

    @SerializedName("ChangeYTD")
    private double changeYTD;

    @SerializedName("ChangePercentYTD")
    private double changePercentYTD;

    @SerializedName("High")
    private double high;

    @SerializedName("Low")
    private double low;

    @SerializedName("Open")
    private double open;
}

File: app/src/main/java/com/bignerdranch/stockwatcher/model/service/StockInfoResponse.kt
package com.bignerdranch.stockwatcher.model.service

import com.squareup.moshi.Json

data class StockInfoResponse(
        @Json(name = "Status") val status: String,
        @Json(name = "Name") val name: String,
        @Json(name = "LastPrice") val lastPrice: String,
        @Json(name = "Change") val change: Double,
        @Json(name = "ChangePercent") val changePercent: Double,
        @Json(name = "MarketCap") val marketCap: Long,
        @Json(name = "Volume") val volume: Long,
        @Json(name = "ChangeYTD") val changeYTD: Double,
        @Json(name = "ChangePercentYTD") val changePercentYTD: Double,
        @Json(name = "High") val high: Double,
        @Json(name = "Low") val low: Double,
        @Json(name = "Open") val open: Double
)

